### ISSUE TITLE
Random failures when running tests in parallel.

### ISSUE THREAD

**Issue by vsespb** [pjcj/Devel--Cover#73]
Hello.

I assume #69 is about Devel::Cover own testsuite.

And I start getting failures when running my own test suite under Devel::Cover.

my code looks like

``` perl
my $harness = TAP::Harness->new({
    formatter_class => 'TAP::Formatter::Console',
    ($ENV{MT_COVER}) ? (switches => $ENV{MT_COVER}) : (exec => [$Config{'perlpath'}]),
    merge           => 1,
    color           => 1,
    jobs            => 8,
});
```

...

``` perl
die unless $harness->runtests(@first, @others)->get_status eq 'PASS';
```

and MT_COVER is `-MDevel::Cover=-db,/tmp/cover_db`

errors are:

```
Can't read /tmp/cover_db/digests with JSON: garbage after JSON object, at character offset 8223 (before "/MtAws/ParentWorker....") at /usr/local/lib/perl/5.14.2/Devel/Cover/DB/IO/JSON.pm line 43.
BEGIN failed--compilation aborted at /home/prj/mt/src/t/unit/metadata.t line 23.
Devel::Cover: Oops, it looks like something went wrong writing the coverage.
              It's possible that more bad things may happen but we'll try to
              carry on anyway as if nothing happened.  At a minimum you'll
              probably find that you are missing coverage.  If you're
              interested, the problem was:

Can't call method "write" on an undefined value at /usr/local/lib/perl/5.14.2/Devel/Cover.pm line 847.

```

(in different files)

perl 5.14.2 Ubuntu, Devel::Cover 1.08

I did not see such errors before (for same testsuite), it started after I installed new OS with perl 5.14 and recent Devel::Cover version.

**Comment by dolmen** [comment]
I'm also affected by this issue when running the testsuite of my project with parrallel testing (prove -j 9).

**Comment by christopherraa** [comment]
Also affected by this error when running the tests in parallel. Perl 5.14.4 and Devel::Cover 1.15 .

**Comment by pjcj** [comment]
I think this should now be fixed as described in #90.

Thanks very much to everyone involved in the ticket!
